src/test/java/org/ays/auth/controller/AysRoleEndToEndTest.java

```diff
@@ -35,9 +37,37 @@ class AysRoleEndToEndTest extends AysEndToEndTest {
     private AysRoleReadPort roleReadPort;
 
 
+    private final AysRoleToRolesSummaryResponseMapper roleToRolesSummaryResponseMapper = AysRoleToRolesSummaryResponseMapper.initialize();
+
+
     private static final String BASE_PATH = "/api/v1";
 
 
+    @Test
+    void whenRolesFound_thenReturnRoles() throws Exception {
+
+        // Initialize
+        Set<AysRole> roles = roleReadPort.findAll();
+
+        // Then
+        String endpoint = BASE_PATH.concat("/roles/summary");
+        MockHttpServletRequestBuilder mockHttpServletRequestBuilder = AysMockMvcRequestBuilders
+                .get(endpoint, adminToken.getAccessToken());
+
+        List<AysRolesSummaryResponse> rolesSummaryResponses = roleToRolesSummaryResponseMapper.map(roles);
+        AysResponse<List<AysRolesSummaryResponse>> response = AysResponse
+                .successOf(rolesSummaryResponses);
+
+        aysMockMvc.perform(mockHttpServletRequestBuilder, response)
+                .andExpect(AysMockResultMatchersBuilders.status()
+                        .isOk())
+                .andExpect(AysMockResultMatchersBuilders.response()
+                        .isNotEmpty())
+                .andExpect(AysMockResultMatchersBuilders.response("size()")
+                        .value(response.getResponse().size()));
```

Comment:
`AysResponse` içerisindeki `response` field'ı liste olmadığı için size kontrolüne ihtiyacımız yok. Burada `AysRolesSummaryResponse` içerisinden gelen field'ların varlığını kontrol etmemiz daha doğru olur. `id` ve `name`'in kesinlikle dolu gelmesi gerekir.

```suggestion
```

Reply:
değişikliği yaptım